integration-tests/test_client_options.py

```diff
@@ -233,16 +233,17 @@ def test_client_diagnosis(insights_client):
         4. Verify the machine id in the diagnostic data matches the
             system's machine id
     :expectedresults:
-        1. The command fails with an error message indicating that diagnosis
-            data cannot be retrieved (404)
+        1. The command fails with a return code of 1 and output includes
+            message 'Could not get diagnosis data.'
         2. The client is registered
         3. The command retrieves diagnostic data and the output contains
             machine id
         4. The machine ID in the diagnostic data matches the system's machine id
     """
     # Running diagnosis on unregistered system returns appropriate error message
     diagnosis_result = insights_client.run("--diagnosis", check=False)
-    assert "Unable to get diagnosis data: 404" in diagnosis_result.stdout
+    assert diagnosis_result.returncode == 1
+    assert "Could not get diagnosis data." in diagnosis_result.stdout
```

Comment:
We probably need to add `if` for the Core version just like we have on other places. This would fail for the RPM Core.